qui/tray/domains.py

```diff
@@ -16,11 +16,21 @@
 import qui.decorators
 import gi  # isort:skip
 gi.require_version('Gtk', '3.0')  # isort:skip
-from gi.repository import Gio, Gtk  # isort:skip
+from gi.repository import Gio, Gtk, GObject  # isort:skip
 
 import gbulb
 gbulb.install()
 
+icons = {}  # this dictionary holds Gtk icons to avoid creating them unnecessarily and thus causing lag
```

Comment:
Be careful with keeping Gtk objects global. Gtk object possibly outliving GtkApplication instance is a bad idea, especially for tests. If there is no better place for it, maybe keep it in some upper level object and pass to the constructor?